utils.go

```diff
@@ -0,0 +1,142 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rabbitmq
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/rabbitmq/amqp091-go"
+)
+
+type Position struct {
+	DeliveryTag  uint64 `json:"deliveryTag"`
+	QueueName    string `json:"queueName"`
+	ExchangeName string `json:"exchangeName"`
+	RoutingKey   string `json:"routingKey"`
+}
+
+func (p Position) ToSdkPosition() sdk.Position {
+	bs, err := json.Marshal(p)
+	if err != nil {
+		// this error should not be possible
+		panic(fmt.Errorf("error marshaling position to JSON: %w", err))
+	}
+
+	return sdk.Position(bs)
+}
+
+func parsePosition(pos sdk.Position) (Position, error) {
+	var p Position
+	err := json.Unmarshal([]byte(pos), &p)
+	if err != nil {
+		return p, fmt.Errorf("error unmarshaling position from JSON: %w", err)
+	}
+
+	return p, nil
+}
+
+func metadataFromMessage(msg amqp091.Delivery) sdk.Metadata {
+	metadata := sdk.Metadata{}
+
+	setKey := func(key string, v any) {
+		if s, ok := v.(string); ok {
+			metadata[key] = s
+			return
+		}
+
+		metadata[key] = fmt.Sprintf("%v", v)
+	}
+
+	setKey("rabbitmq.queueName", msg.MessageCount)
```

Comment:
aren't all these values already included in the msg sent to the destination? if not, then ignore this 👀

Reply:
Some of them are encoded into the sdk.Position. I guess that makes them redundant?

For the other parameters, I don't really know which ones are relevant to conduit and which aren't, so to be sure, I included all props that I found in the `amqp091.Delivery` struct.

@maha-hajja thoughts on this?